Automattic/wp-calypso
# Help: Refactor support article alternates to use `react-query`

Support article alternates are currently used to fetch localized support articles for non-English locales. The idea is that in the few areas where we use inline support articles, we'd like to serve the articles in the locale of the current user. 

That area of the code previously used redux, however much of the implementation is boilerplate code and parts of it are even not used at all. 

This represents a good opportunity to refactor to `react-query`, and it's also demonstrated by the net lines of code of this PR.  

#### Changes proposed in this Pull Request

* Introduce a new `useSupportArticleAlternatesQuery` query.
* Update `SupportArticleDialog` to use `react-query`.
* Update `SupportArticleDialog` to use hooks instead of `connect` - previously it was a mix of `connect()` and hooks, and that was making refactoring difficult.
* Update `InlineSupportLink` to prefetch the react query and simplify by removing the arbitrary component state.
* Remove the legacy redux state and query component.

#### Testing instructions

Note: the crossed-out instructions are no longer relevant since we removed the prefetching in one of the iterations.

* Keep your dev tools open, particularly the Network tab.
* Test with the default (English) locale, where this functionality shouldn't be triggered:
  * Start as a user with **English** set as interface language in Calypso.
  * Go to `/home/:site`. 
  * Below the stats, there's a slider and the first slide says **"Blog like an expert from day one"**. Navigate to the second slide.
  * ~Roll over the **"Learn more"** link.~
  * ~Verify that a request to `/support/alternates/...` is NOT performed in the background.~
  * Click the **"Learn more"** link.
  * Verify that the **"Free Photo Library (Pexels)"** article is loaded correctly in the dialog, and a `?support-article=145498` query arg is added to the URL.
  * Verify that a request to `/support/alternates/...` is NOT performed.
  * Refresh the page and verify that the dialog gets open automatically and still loads correctly.
  * Click outside the dialog or on the **"Close"** button and verify it's correctly closed.
  * Navigate to `/me/account`.
* Test with another locale, where this article is translated and the functionality should be triggered:
  * Change your interface language to **Español**.
  * Go to `/home/:site`. 
  * Below the stats, there's a slider and the first slide says **"Sé un bloguero experto desde el primer día"**. Navigate to the second slide.
  * ~Roll over the **"Más información"** link.~
  * ~Verify that a request to `/support/alternates/...` is performed in the background.~
  * Click the **"Más información"** link.
  * Verify that the **"Biblioteca de fotos gratuitas (Pexels)"** article is loaded correctly in the dialog, and a `?support-article=145498` query arg is added to the URL.
  * Verify that a request to `/support/alternates/...` was performed.
  * Refresh the page and verify that the dialog gets open automatically and still loads correctly.
  * Click outside the dialog or on the **"Cerrar"** button and verify it's correctly closed.
* Test with another locale, where this article is not translated and the functionality should be triggered but should fall back to English:
  * Change your interface language to **Български** (Bulgarian).
  * Go to `/home/:site`. 
  * Below the stats, there's a slider and the first slide says **"Blog like an expert from day one"**. Navigate to the second slide.
  * ~Roll over the **"Научете повече"** link.~
  * ~Verify that a request to `/support/alternates/...` is performed in the background.~
  * Click the **"Научете повече"** link.
  * Verify that the **"Free Photo Library (Pexels)"** article is loaded correctly in the dialog, and a `?support-article=145498` query arg is added to the URL. 
  * Verify the article is in English (a Bulgarian translation doesn't exist).
  * Verify that a request to `/support/alternates/...` is performed.
  * Refresh the page and verify that the dialog gets open automatically and still loads correctly.
  * Click outside the dialog or on the **"Close"** button and verify it's correctly closed.
* Verify that the removed code is not in use.
